tardis/model/base.py

```diff
@@ -206,77 +113,25 @@ class SimulationState(HDFWriterMixin):
     def __init__(
         self,
         geometry,
-        density,
-        abundance,
-        isotope_abundance,
+        composition,
         time_explosion,
         t_inner,
-        elemental_mass,
         luminosity_requested=None,
         t_radiative=None,
         dilution_factor=None,
         electron_densities=None,
     ):
         self.geometry = geometry
+        self.composition = composition
 
-        self._abundance = abundance
         self.time_explosion = time_explosion
         self._electron_densities = electron_densities
 
-        if len(density) != len(self.geometry.v_inner_active):
-            density = density[
+        if len(composition.density) != len(self.geometry.v_inner_active):
+            density = composition.density[
                 self.geometry.v_inner_boundary_index : self.geometry.v_outer_boundary_index
             ]
 
-        self.raw_abundance = self._abundance
-        self.raw_isotope_abundance = isotope_abundance
-
-        atomic_mass = None
-        if elemental_mass is not None:
-            mass = {}
-            stable_atomic_numbers = self.raw_abundance.index.to_list()
-            for z in stable_atomic_numbers:
-                mass[z] = [
-                    elemental_mass[z]
-                    for i in range(self.raw_abundance.columns.size)
-                ]
-            stable_isotope_mass = pd.DataFrame(mass).T
-
-            isotope_mass = {}
-            for atomic_number, i in self.raw_isotope_abundance.decay(
```

Comment:
@AlexHls @andrewfullard this was added in PR https://github.com/tardis-sn/tardis/pull/2118 - which was signed by both of you. I don't quite understand what this code does. I think something like effective element_masses  but I don't know - can you check.

Reply:
I think this just gets the total mass of each element, separated by the stable and radioactive isotopes. Where or why this is needed I cannot tell.

Reply:
This was/is almost certainly something I requested to deal with the issues around isotope vs atomic masses, as support for the gamma-ray code. It means that the final atomic mass of the composition is a collection of stable and unstable isotopes that have been decayed to the appropriate time.

Reply:
and the concat is a bit unusual because it means if there is stable isotopes and non stable isotopes in the same composition then it fails - afaict?